### PR TITLE
Made minimum time for health update to 5 seconds. Added more logging

### PR DESCRIPTION
**Pull request by r0goyal**

### CODE REVIEW

File: src/main/java/com/flipkart/ranger/serviceprovider/ServiceProviderBuilder.java [hunk 1]
```diff
@@ -134,24 +134,37 @@
...
+
+        if (staleUpdateThresholdMs < 5000) {
+            LOGGER.warn("Stale update threshold too low (should be between 5000ms and 20000ms). Current value: {} ms. " +
+                    "Being set to 5000ms", staleUpdateThresholdMs);
+            staleUpdateThresholdMs = 5000;
         }
+        else if (staleUpdateThresholdMs > 20000) {
+            LOGGER.warn("Stale update threshold too high (should be between 5000ms and 20000ms). Current value: {} ms. " +
+                    "Being set to 45000ms", staleUpdateThresholdMs);
+            staleUpdateThresholdMs = 20000;
```
Comment (santanusinha) [inline src/main/java/com/flipkart/ranger/serviceprovider/ServiceProviderBuilder.java]: Set default to one value

File: src/main/java/com/flipkart/ranger/serviceprovider/ServiceProviderBuilder.java [hunk 2]
```diff
@@ -134,24 +134,37 @@
...
+
+        if (healthUpdateIntervalMs < 1000) {
+            LOGGER.warn("Health update interval too low (should be between 1000ms and 20000ms). Current value: {} ms. " +
+                    "Being set to 1000ms", healthUpdateIntervalMs);
+            healthUpdateIntervalMs = 1000;
+        }
+        else if (healthUpdateIntervalMs > 20000) {
+            LOGGER.warn("Health update interval too high (should be between 1000ms and 20000ms). Current value: {} ms. " +
+                    "Being set to 20000ms", healthUpdateIntervalMs);
+            healthUpdateIntervalMs = 20000;
```
Comment (santanusinha) [inline src/main/java/com/flipkart/ranger/serviceprovider/ServiceProviderBuilder.java]: Set default to one value

File: src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java [hunk 3]
```diff
@@ -109,26 +109,32 @@ public void checkForUpdate() {
...
             final long healthcheckZombieCheckThresholdTime = System.currentTimeMillis() - 60000; //1 Minute
             final Service service = serviceRegistry.getService();
+            final String serviceName = service.getServiceName();
             if(!service.isRunning()) {
                 return Optional.empty();
             }
             final Deserializer<T> deserializer = serviceRegistry.getDeserializer();
             final CuratorFramework curatorFramework = service.getCuratorFramework();
             final String parentPath = PathBuilder.path(service);
+            logger.info("Looking for node list of [{}]", serviceName);
```
Comment (santanusinha) [inline src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java]: debug

File: src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java [hunk 4]
```diff
@@ -109,26 +109,32 @@ public void checkForUpdate() {
...
             List<String> children = curatorFramework.getChildren().forPath(parentPath);
             List<ServiceNode<T>> nodes = Lists.newArrayListWithCapacity(children.size());
+            logger.info("Found {} nodes for [{}]", nodes.size(), serviceName);
```
Comment (santanusinha) [inline src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java]: debug

File: src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java [hunk 5]
```diff
@@ -139,14 +145,15 @@ public void checkForUpdate() {
     }
 
     public void stop() {
-        logger.debug("Stopped updater");
+        logger.info("Stopped updater for [{}]", serviceRegistry.getService().getServiceName());
```
Comment (santanusinha) [inline src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java]: debug

File: src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java [hunk 6]
```diff
@@ -139,14 +145,15 @@ public void checkForUpdate() {
...
     }
 
 
     private void updateRegistry() {
         List<ServiceNode<T>> nodes = checkForUpdateOnZookeeper().orElse(null);
         if(null != nodes) {
-            logger.debug("Setting nodelist of size: {}", nodes.size());
+            logger.info("Updating nodelist of size: {} for [{}]", nodes.size(),
```
Comment (santanusinha) [inline src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java]: debug

File: src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java (hunk 5 above)
Comment (r0goyal) [reply src/main/java/com/flipkart/ranger/finder/ServiceRegistryUpdater.java]: This is done once during the lifetime of registry. Corresponding start method also has similar message at INFO level

**Pull request merged**